### PR TITLE
feat: add geoip support to sentry deployment

### PR DESCRIPTION
**Pull request by patsevanton**
This pull request introduces support for GeoIP in the Sentry deployment. The changes include:

1. **New Configuration Options**:
   - Added new configuration options for GeoIP in the `values.yaml` file.
   - The new options include `accountID`, `licenseKey`, `editionIDs`, `persistence`, `volumeName`, `mountPath`, and `path`.

2. **New Templates**:
   - Added a new `deployment-geoip-job.yaml` template to handle the GeoIP database update job.
   - Added a new `pvc-geoip.yaml` template for the PersistentVolumeClaim (PVC) to store the GeoIP database.
   - Added a new `secret-geoip-env.yaml` template to manage the environment variables required for GeoIP configuration.

3. **Volume Mounts**:
   - Added volume mounts for the GeoIP database in the `deployment-relay.yaml`, `deployment-sentry-web.yaml`, and `deployment-sentry-worker-events.yaml` templates.

4. **Documentation**:
   - Updated the `README.md` file with an example of how to configure GeoIP in the Sentry deployment.

### CODE REVIEW

**Review by Mokto — APPROVED** [review]

**Comment by swade1987** [comment]
We should allow the user to provide the GeoIP configuration via a secret. Otherwise, we are adding tech-debt like we had when being unable to provide the Redis password via a secret. A license key is not something that should be stored in plain text anywhere as its a "secret".

**Comment by patsevanton** [comment]
@swade1987  You want to instead of indicating
```
geodata:
  licenseKey: “licenseKey”.
```  
specify
```
geodata:
  licenseKey: secret name with licenseKey?
```

**Comment by swade1987** [comment]
> @swade1987 You want to instead of indicating
> 
> ```
> geodata:
>   licenseKey: “licenseKey”.
> ```
> 
> specify
> 
> ```
> geodata:
>   licenseKey: secret name with licenseKey?
> ```

All the configuration options for the geodata could be passed as a single secret. 
I would typically say `ConfigMap` but as the license key is a secret that outweighs it in this instance.

File: charts/sentry/templates/pvc-geoip.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  accessModes:
+    - ReadWriteMany
+  resources:
+    requests:
+      storage: {{ .Values.geodata.persistence.size }}
+{{- if (eq "-" .Values.geodata.persistence.storageClass) }}
+  storageClassName: ""
+{{- else }}
+  storageClassName: "{{ .Values.geodata.persistence.storageClass }}"
+  {{- end }}
```
Comment (sdernbach-ionos) [inline charts/sentry/templates/pvc-geoip.yaml:21]: @patsevanton according to the comment in the values.yaml shouldn't that be:

```
{{- if .Values.geodata.persistence.storageClass }}
{{- if (eq "-" .Values.geodata.persistence.storageClass) }}
  storageClassName: ""
{{- else }}
  storageClassName: "{{ .Values.geodata.persistence.storageClass }}"
{{- end }}
{{- end }}
```

Comment (patsevanton) [reply charts/sentry/templates/pvc-geoip.yaml:21]: @sdernbach-ionos please review PR https://github.com/sentry-kubernetes/charts/pull/1524

**Pull request merged**